File: androidsupport/src/main/java/com/tq/libs/common/helper/FileHelper.java
/*
 * ******************************************************************************
 *  Copyright Ⓒ 2016. TrinhQuan. All right reversed
 *  Author: TrinhQuan. Created on 2016/10/15
 *  Contact: trinhquan.171093@gmail.com
 * ******************************************************************************
 */

package com.tq.libs.common.helper;

import android.content.Context;
import android.support.annotation.CheckResult;

import java.io.File;

public class FileHelper {

    public static File getCachedFolder(Context context) {
        File cachedFolder = context.getExternalCacheDir();
        if (cachedFolder == null) {
            cachedFolder = context.getCacheDir();
        }
        return cachedFolder;
    }

    public static File getAppFolder(Context context) {
        File appFolder = context.getExternalFilesDir(null);
        if (appFolder == null) {
            appFolder = context.getFilesDir();
        }
        return appFolder;
    }
}

File: kotlinandroidsupport/src/main/java/com/tq/klibs/ext/FileHelper.kt
/*
 * ******************************************************************************
 *  Copyright Ⓒ 2016. TrinhQuan. All right reversed
 *  Author: TrinhQuan. Created on 2016/10/15
 *  Contact: trinhquan.171093@gmail.com
 * ******************************************************************************
 */

package com.tq.klibs.ext

import android.content.Context

import java.io.File

fun Context.getCachedFolder(): File {
    var cachedFolder = this.externalCacheDir
    if (cachedFolder == null) {
        cachedFolder = this.cacheDir
    }
    return cachedFolder
}

fun Context.getAppFolder(): File {
    var appFolder = this.getExternalFilesDir(null)
    if (appFolder == null) {
        appFolder = this.filesDir
    }
    return appFolder
}